src/Text/SlamSearch.purs

```diff
@@ -0,0 +1,36 @@
+module Text.SlamSearch where
+
+
+import Data.Semiring.Free
+import Data.String
+import Data.Foldable
+import Data.Either
+import Data.Array (reverse)
+import qualified Data.Semiring.Disjunctive as Dj
+
+import Text.SlamSearch.Types
+import qualified Text.SlamSearch.Parser.Tokens as Tk
+import qualified Text.SlamSearch.Parser as S
+import qualified Text.Parsing.Parser as P
+
+
+mkQuery :: String -> SearchQuery
+mkQuery input =
+  liftFree mkTerm $ prepare input
+  where prepare :: String -> Free String
+        prepare input = foldl (*) one $ free <$> reverse (split " " input)
+
+        
+        mkTerm :: String -> SearchQuery
+        mkTerm input =
+          either (const one) free $ parseTerm input
```

Comment:
They fold by `*`, that's because I've put `one` here. 
Yes, parse errors are thrown away, string that have one correct query and one incorrect will be parsed to `Free` with one correct `Term`.